src/libcore/iter.rs

```diff
@@ -200,7 +147,7 @@ pub trait IteratorExt: Iterator + Sized {
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
-    fn nth(&mut self, mut n: usize) -> Option<Self::Item> {
+    fn nth(&mut self, mut n: usize) -> Option<Self::Item> where Self: Sized {
```

Comment:
I think that functions like this which *can* relax the `Self: Sized` bound are ok to. In theory an iterator could override this method to provide a specialized implementation.